src/System.Management.Automation/engine/Utils.cs

```diff
@@ -990,7 +990,32 @@ internal static bool NativeItemExists(string path, out bool isDirectory, out Exc
             if (result == -1)
             {
                 int errorCode = Marshal.GetLastWin32Error();
-                if (errorCode == 5)
+
+                // Errorcode 32 is 'ERROR_SHARING_VIOLATION' i.e. 
+                // The process cannot access the file because it is being used by another process.
+                // GetFileAttributes will return INVALID_FILE_ATTRIBUTES for a system file or directory because of this error.
+                // Fallback to FindFirstFile to check if the file actually exists or not.
+                if (errorCode == 32)
```

Comment:
And another one -- not related to this fix -- it seems for any other error code we just silently sallow it without throw error. Probably it was by design. It might be a breaking change to fix that.